### PR TITLE
Fix pool actives append error when NIO.NIOConnectionError happened

### PR DESCRIPTION
**Pull request by vkill**
Hi,

I restart redis server after app startup, then all actives are unavailable and no connection.

Because NIO.NIOConnectionError will happened when call database.newConnection.

### CODE REVIEW

File: Sources/DatabaseKit/ConnectionPool/DatabaseConnectionPool.swift [hunk 1]
```diff
@@ -58,7 +59,13 @@ public final class DatabaseConnectionPool<Database> where Database: DatabaseKit.
...
         let release = releaseConnection
         return requestConnection().flatMap(to: T.self) { conn in
-            return try closure(conn).always { release(conn) }
+            do {
+                return try closure(conn).always { release(conn) }
+            } catch {
+                release(conn)
+
+                throw error
+            }
```
Comment (fpillet) [inline Sources/DatabaseKit/ConnectionPool/DatabaseConnectionPool.swift:68]: this fix is also required in `Container+NewConnection.swift`, here https://github.com/vapor/database-kit/blob/master/Sources/DatabaseKit/Connection/Container%2BNewConnection.swift#L20

Without it, the connection is never released if `closure` throws.

Comment (vkill) [reply Sources/DatabaseKit/ConnectionPool/DatabaseConnectionPool.swift:68]: Yes, I have fixed it.

**Comment by fpillet** [comment]
Ping @tanner0101 : these fixes for releasing DB connections are really important. 👍

**Review by tanner0101 — APPROVED** [review]
This change LGTM, thanks for the patch @vkill. Also, thanks for the ping @fpillet. This one slipped through my notifications.

**Comment by penny-coin** [comment]
Hey @vkill, you just merged a pull request, have a coin! 

You now have 4 coins.

**Pull request merged**